tee-worker/omni-executor/intent/asset-lock/src/account_wide.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright 2020-2024 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+use std::sync::RwLock;
+
+use crate::AssetsLock;
+
+// Manages whether any asset is locked for particular account or not.
+pub struct AccountWideAssetsLock {
+	locked: RwLock<bool>,
```

Comment:
How about using `std::sync::atomic::AtomicBool` here instead? We can avoid the overhead of acquiring and releasing a lock and the code would get simpler, e.g: 

```rust
self.locked.swap(true, Ordering::SeqCst)
...
self.locked.swap(false, Ordering::SeqCst)
```

Reply:
I've changed it to plain `bool` .